framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeMirror.java

```diff
@@ -1173,13 +1173,15 @@ public void addAnnotation(AnnotationMirror a) {
         /**
          * Sets the parameter types of this executable type, excluding the receiver.
          *
-         * @param params the parameter types, excluding the receiver
+         * @param params an unmodifiable list of parameter types to be captured by this method,
+         *     excluding the receiver
          */
-        /*package-private*/ void setParameterTypes(List<? extends AnnotatedTypeMirror> params) {
-            paramTypes =
-                    params.isEmpty()
-                            ? Collections.emptyList()
-                            : Collections.unmodifiableList(new ArrayList<>(params));
+        /*package-private*/ void setParameterTypes(List<AnnotatedTypeMirror> params) {
+            if (params.isEmpty()) {
```

Comment:
We always need to check whether a list is empty or not, either at the call site or in the method. I think it's simpler to leave the check here unless we are confident that future changes will always follow the same rule.

Reply:
Since the upstream (typetools) implements this method in a more conservative way, should we expect future changes to always:
1. Use `Collections.emptyList()` when the list should be empty
2. Pass a list which won't be modified

Reply:
Sorry, didn't see your comments before I started editing.
I don't know why some changes in master are showing up in this PR...
Checking once at the call site seems easier, as we control all call sites. What do you think of the changes I made?

Reply:
Thanks! I think the changes look good to me. What I'm worried about is how to maintain this pattern when we pull from the upstream.